### PR TITLE
luanaruiz9 pcfg example

### PR DESCRIPTION
**Pull request by luanaruiz9**
Updated model files to process PCFG data in its original form. The main changes were the addition of functions to calculate sentence accuracy from both logits and predictions (beam_size=4). I have also removed/modified a few hyperparameters specific to machine translation (e.g., the reverse translation flag was removed, and the brevity penalty parameter alpha was set to zero).

### CODE REVIEW

File: model/configs/default.py [hunk 1]
```diff
@@ -25,39 +25,40 @@ def get_config():
...
 
-  # Name of TFDS translation dataset to use.
-  config.dataset_name = "wmt17_translate/de-en"
+  # Name of TFDS PCFG dataset to use.
+  config.dataset_name = "pcfg_full_data_original"
 
-  # Optional name of TFDS translation dataset to use for evaluation.
-  config.eval_dataset_name = "wmt14_translate/de-en"
+  # Optional name of TFDS PCFG dataset to use for evaluation.
+  config.eval_dataset_name = False
```
Comment (jtainslie) [inline model/configs/default.py]: Nit: Maybe `None` would be more typical if this is an optional string field.

File: model/train.py [hunk 2]
```diff
@@ -161,22 +161,48 @@ def compute_weighted_accuracy(logits, targets, weights=None):
     normalizing_factor = weights.sum()
 
   return loss.sum(), normalizing_factor
+  
+  
+def compute_sentence_accuracy(logits, targets, weights=None):
+  """Compute sentence accuracy for log probs and targets.
```
Comment (jtainslie) [inline model/train.py]: For the style guide, docstrings should be descriptive-style rather than imperative-style.  (https://google.github.io/styleguide/pyguide.html#383-functions-and-methods)
```suggestion
  """Computes sentence accuracy for log probs and targets.
```

File: model/train.py [hunk 3]
```diff
@@ -161,22 +161,48 @@ def compute_weighted_accuracy(logits, targets, weights=None):
...
+  Args:
+   logits: [batch, length, num_classes] float array.
+   targets: categorical targets [batch, length] int array.
+   weights: None or array of shape [batch, length]
+
+  Returns:
+    Tuple of scalar loss and batch normalizing factor.
+  """
+  if logits.ndim != targets.ndim + 1:
+    raise ValueError("Incorrect shapes. Got shape %s logits and %s targets" %
```
Comment (jtainslie) [inline model/train.py]: Nit: you could use formatted string literals instead (https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals).

File: model/train.py [hunk 4]
```diff
@@ -161,22 +161,48 @@ def compute_weighted_accuracy(logits, targets, weights=None):
...
+                     (str(logits.shape), str(targets.shape)))
+  boolean_array = jnp.equal(jnp.argmax(logits, axis=-1), targets)
+  normalizing_factor = logits.shape[0]
+  if weights is not None:
+    boolean_array = jnp.equal(weights * jnp.argmax(logits, axis=-1), weights * targets)
+    normalizing_factor = jnp.sum(jnp.sum(weights, axis=-1)>0)
```
Comment (jtainslie) [inline model/train.py]: Nit: spacing
```suggestion
    normalizing_factor = jnp.sum(jnp.sum(weights, axis=-1) > 0)
```

File: model/train.py [hunk 5]
```diff
@@ -571,13 +604,38 @@ def decode_tokens(toks):
           lr = train_metrics.pop("learning_rate").mean()
           metrics_sums = jax.tree_map(jnp.sum, train_metrics)
           denominator = metrics_sums.pop("denominator")
+          sentence_acc = metrics_sums.pop("sentence_accuracy")
+          sentence_denominator = metrics_sums.pop("sentence_denominator")
           summary = jax.tree_map(lambda x: x / denominator, metrics_sums)  # pylint: disable=cell-var-from-loop
+          summary["sentence_accuracy"] = sentence_acc/sentence_denominator
```
Comment (jtainslie) [inline model/train.py]: Nit: spacing.
```suggestion
          summary["sentence_accuracy"] = sentence_acc / sentence_denominator
```

**Review by jtainslie — APPROVED** [review]
Lgtm apart from minor nits.

**Review by santiaontanon-google — APPROVED** [review]
